### ISSUE TITLE
ctx.activePlayers not updated after triggering endStage event during player's move

### ISSUE THREAD

**Issue by JanMolcik** [boardgameio/boardgame.io#835]
I am trying to implement this functionality:

- player plays a card
- player draws a card
- after draw the turn ends automatically 

_But_ certain cards enable the player to repeat the play card and draw card sequence without ending turn after the draw stage.

What I wanted to achieve here was checking for this certain cards and if they are played I trigger the `setActivePlayers` event with the `next` property like this:

`ctx.events.setActivePlayers({ currentPlayer: "draw", next: { currentPlayer: "play" } })`.

The issue I am running into is that when I trigger endStage in the draw card move and then check if I should also trigger endTurn (the workaround adviced in [this issue](https://github.com/boardgameio/boardgame.io/issues/769#issuecomment-667664528)) 
```
ctx.events.endStage();
console.log("activePlayers: ", ctx.activePlayers);
if (!ctx.activePlayers) {
    ctx.events.endTurn();
}
```
the setActivePlayers seems to have old state even after the endStage event was already triggered. I setup simple code sample in [codesandbox](https://codesandbox.io/s/bgio-endstage-isuxj) to clarify my approach.

I would really appreciate any suggestions on how to accomplish this functionality or any workarounds that I can make.

**Comment by JanMolcik** [comment]
I found a workaround for this case. Since I only need to repeat the sequence if there is `next` activePlayers object I just check `ctx._nextActivePlayers` within the endTurn check

```
ctx.events.endStage();

if (!ctx.activePlayers || !ctx._nextActivePlayers) {
    ctx.events.endTurn();
}
```

However, I feel like this behaviour could use some investigation unless I am missing something. I will leave this open and let you decide whether to close it.

**Comment by delucis** [comment]
Hi @JanMolcik — glad you’ve got it working.

As mentioned [in the docs](https://boardgame.io/documentation/#/events?id=triggering-an-event-from-game-logic):

> Events are queued up and triggered **after** a move. Any changes you make to `G` will be applied before events are triggered, even if the event is called first in your move function.

So `endStage` will always take effect after your checks here, because it is only processed once your move is finished.

In theory, the idiomatic way to implement the behaviour you want is to use a `turn.endIf` hook to check if `activePlayers` is empty, but as you saw in #769, there is currently a bug preventing that approach, so your solution is probably the best currently possible.

I’ll close this as `activePlayers` is not expected to update *during* a move and it sounds like the best way to handle this would be to fix #769.

**Comment by JanMolcik** [comment]
Thanks @delucis for clarifying. I must have missed it and it absolutely makes sense now!
